storage/http_client.go

```diff
@@ -684,7 +684,56 @@ func (c *httpStorageClient) ComposeObject(ctx context.Context, req *composeObjec
 	return newObject(obj), nil
 }
 func (c *httpStorageClient) RewriteObject(ctx context.Context, req *rewriteObjectRequest, opts ...storageOption) (*rewriteObjectResponse, error) {
-	return nil, errMethodNotSupported
+	s := callSettings(c.settings, opts...)
+	rawObject := req.attrs.toRawObject("")
+	call := c.raw.Objects.Rewrite(req.srcBucket, req.srcObject, req.dstBucket, req.dstObject, rawObject)
+
+	call.Context(ctx).Projection("full")
+	if req.token != "" {
+		call.RewriteToken(req.token)
+	}
+	if req.dstKeyName != "" {
+		call.DestinationKmsKeyName(req.dstKeyName)
+	}
+	if req.predefinedACL != "" {
+		call.DestinationPredefinedAcl(req.predefinedACL)
+	}
+	if err := applyConds("Copy destination", req.gen, req.conds, call); err != nil {
+		return nil, err
+	}
+	if err := applySourceConds(req.srcGen, req.srcConds, call); err != nil {
+		return nil, err
+	}
+	// The userProject, whether source or destination project, is decided by the code calling the interface.
```

Comment:
UserProject is just about billing for the call; doesn't have anything to do with source/destination. (Though in the current code it's set via the source or destination object handles, which is confusing).

Reply:
I'll just remove the comment then.